### ISSUE TITLE
iPhone videos discolored

### ISSUE THREAD

**Issue by vishnukvmd** [ente-io/ente#1166]
### Description

[Migrated from Discord]

> I am currently on Android but I used to have an iPhone and I noticed that all of my videos from iOS are discolored when I view them on ente. They looked fine from the iOS Photos app but from ente they are washed out. It's like the saturation and vibrancy were turned way down. I'm wondering if there is a setting somewhere that should have been changed. This was not exclusive to one phone, I noticed this on both the iPhone 13 pro and iPhone 14 pro.

This could be codec related, since a different customer reported that videos recorded in compatibility mode turned out fine.

### What product are you using?

Ente Photos

### What platform are you using?

Mobile - iOS

**Comment by vishnukvmd** [comment]
Potential root cause: https://github.com/flutter/flutter/issues/91241

**Comment by mertalev** [comment]
Yup, that sounds like the same issue. Immich has the same problem since we also use Flutter. IIRC it's because Flutter interacts with the native video player through a texture rather than a PlatformView. The former doesn't allow for HDR while the latter does.

As for alternatives, there are players like media_kit that tone-map videos so they look normal, albeit not HDR. There's also a [stale branch](https://github.com/flutter/flutter/issues/86613) to switch the Flutter video player to use PlatformView. But I don't think there are currently any actively-maintained packages that play videos in HDR on iOS.
